### PR TITLE
EVG-16772 don't set display task start time for exec tasks that have zero start time

### PR DESCRIPTION
**Pull request by hadjri**
[EVG-16772](https://jira.mongodb.org/browse/EVG-16772)

### Description 
Although I was not able to emulate this error in staging I have noticed some logic in `UpdateDisplayTaskForTask` where the display task's start time is set to the lowest value of all of its execution tasks' start times. What seems to be happening is some execution tasks have nil / zero start time set are being detected as the "lowest" start time and updating the display task as such. Small change has been made to not update the display task's start time if the execution task we are inspecting has a nil / zero start time.
### Testing 
Added case to unit test for `UpdateDisplayTaskForTask`

### CODE REVIEW

File: model/task_lifecycle.go [hunk 1]
```diff
@@ -1583,7 +1583,7 @@ func UpdateDisplayTaskForTask(t *task.Task) error {
 		timeTaken += execTask.TimeTaken
 
 		// set the start/end time of the display task as the earliest/latest task
-		if execTask.StartTime.Before(startTime) {
+		if execTask.StartTime != utility.ZeroTime && execTask.StartTime.Before(startTime) {
```
Comment (Kimchelly) [inline model/task_lifecycle.go:1586]: `utility.IsZeroTime(execTask.StartTime)`?

File: model/task_lifecycle_test.go [hunk 2]
```diff
@@ -3723,6 +3723,16 @@ func TestDisplayTaskUpdates(t *testing.T) {
 		},
 	}
 	assert.NoError(dt2.Insert())
+	dt3 := task.Task{
+		Id:          "dt2",
+		DisplayOnly: true,
+		Status:      evergreen.TaskUndispatched,
+		Activated:   false,
+		ExecutionTasks: []string{
+			"task11",
```
Comment (minnakt) [inline model/task_lifecycle_test.go:3732]: Did you mean to add `task12` to the execution tasks? (I don't think it's currently being used anywhere)

File: model/task_lifecycle_test.go [hunk 3]
```diff
@@ -3857,6 +3880,13 @@ func TestDisplayTaskUpdates(t *testing.T) {
 	assert.NoError(err)
 	assert.NotNil(dbTask)
 	assert.Equal(evergreen.TaskFailed, dbTask.Status)
+
+	// a display task should not set its start time to any exec tasks that have zero start time
+	assert.NoError(UpdateDisplayTaskForTask(&task11))
+	dbTask, err = task.FindOne(db.Query(task.ById(dt3.Id)))
+	assert.NoError(err)
+	assert.NotNil(dbTask)
+	assert.Equal(time.Date(2000, 0, 0, 0, 44, 0, 0, time.Local), dbTask.StartTime)
```
Comment (minnakt) [inline model/task_lifecycle_test.go:3889]: maybe `assert.Equal(task12.StartTime, dbTask.StartTime)`?

File: model/task_lifecycle_test.go [hunk 4]
```diff
@@ -3723,6 +3723,16 @@ func TestDisplayTaskUpdates(t *testing.T) {
 		},
 	}
 	assert.NoError(dt2.Insert())
+	dt3 := task.Task{
+		Id:          "dt2",
```
[the remark this answers is not in the document]
Comment (hadjri) [reply model/task_lifecycle_test.go:3727]: Ah good catch

**Review by Kimchelly — APPROVED** [review]

File: model/task_lifecycle_test.go [hunk 5]
```diff
@@ -3857,6 +3881,14 @@ func TestDisplayTaskUpdates(t *testing.T) {
...
 	assert.NotNil(dbTask)
 	assert.Equal(evergreen.TaskFailed, dbTask.Status)
+
+	// a display task should not set its start time to any exec tasks that have zero start time
+	assert.NoError(UpdateDisplayTaskForTask(&task11))
+	dbTask, err = task.FindOne(db.Query(task.ById(dt3.Id)))
+	assert.NoError(err)
+	assert.NotNil(dbTask)
+	assert.Equal(task12.StartTime, dbTask.StartTime)
+	assert.Equal(time.Date(2000, 0, 0, 0, 44, 0, 0, time.Local), dbTask.StartTime)
```
Comment (minnakt) [inline model/task_lifecycle_test.go:3891]: You might have forgotten to remove `L3891`! (`L3890` and `L3891` are identical assertions)

**Review by minnakt — APPROVED** [review]
Thank you for taking care of this bug, now the task duration page will look even better!!! 😇

**Pull request merged**